app/registries/foreman/setting_manager.rb

```diff
@@ -71,6 +79,42 @@ def setting(name, default:, description:, type:, full_name: nil, value: nil, col
           encrypted: encrypted,
           options: options,
         }
+        validates(name, validate) if validate
+        storage[name.to_s]
+      end
+
+      def validates(name, validations)
+        if validations.is_a?(Proc)
+          validates_with name, ValueLambdaValidator, proc: validations
+          return
+        elsif validations.is_a?(Regexp)
+          validations = { format: { with: validations } }
+        elsif validations.is_a?(Symbol)
+          validations = { validations => true }
+        end
+        _add_default_validation_options(name, validations)
+        Setting.validates(:value, validations)
+      end
+
+      def validates_with(name, *args, &block)
+        options = args.extract_options!
+        _add_default_validation_options(name, options)
+        options[:attributes] = [:value]
+        args << options
+        Setting.validates_with(*args, &block)
+      end
+
+      def _add_default_validation_options(setting_name, options)
+        _wrap_validation_if(setting_name, options)
+        options[:allow_blank] = true
```

Comment:
Following our discussion, let's drop this completely and let the user specify it if needed by using `validates` or `validates_with`